### PR TITLE
Use plugins resources instead of registries

### PR DESCRIPTION
**Pull request by Goaman**

### CODE REVIEW

File: addons/html_builder/static/src/builder/core_builder_action_plugin.js [hunk 1]
```diff
@@ -0,0 +1,81 @@
+import { Plugin } from "@html_editor/plugin";
+import { registry } from "@web/core/registry";
+
+class CoreBuilderActionPlugin extends Plugin {
+    static id = "CoreBuilderAction";
+    resources = {
+        builder_actions: Object.entries(actions).map(([id, action]) => ({ id, ...action })),
```
Comment (FrancoisGe) [inline addons/html_builder/static/src/builder/core_builder_action_plugin.js]: why not do this processing in the builder_actions plugin

File: addons/html_builder/static/src/builder/core_builder_action_plugin.js [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+    };
+}
+registry.category("website-plugins").add(CoreBuilderActionPlugin.id, CoreBuilderActionPlugin);
+
+const styleMap = {
```
Comment (FrancoisGe) [inline addons/html_builder/static/src/builder/core_builder_action_plugin.js:12]: It need to be extendable no ?Ressource ?

File: addons/html_builder/static/src/builder/options/content_width_option.js [hunk 3]
```diff
@@ -1,8 +1,7 @@
-import { registry } from "@web/core/registry";
+import { addSimpleOptionPlugin } from "../builder_helpers";
 
-// TODO to import in html_builder
-
-registry.category("sidebar-element-option").add("ContentWidthOption", {
+addSimpleOptionPlugin({
+    pluginId: "ContentWidthOption",
```
Comment (FrancoisGe) [inline addons/html_builder/static/src/builder/options/content_width_option.js]: pluginId or Id ?

OptionComponent => Component ?
Maybe trow error if template + component

File: addons/html_builder/static/src/builder/builder_helpers.js [hunk 4]
```diff
@@ -361,3 +357,19 @@ export const clickableWeWidgetProps = {
     dataAttributeActionValue: { type: [String, Array, validateIsNull], optional: true },
     styleActionValue: { type: [String, Array, validateIsNull], optional: true },
 };
+
+export function addSimpleOptionPlugin({ pluginId, template, selector, sequence, OptionComponent }) {
```
Comment (Goaman) [inline addons/html_builder/static/src/builder/builder_helpers.js]: register

File: addons/html_builder/static/tests/helpers.js [hunk 5]
```diff
@@ -100,29 +102,50 @@ export function getEditable(inWrap) {
...
         selector,
         sequence,
     });
+    registry.category("website-plugins").add(pluginId, Class);
     after(() => {
-        registry.category("sidebar-element-option").remove(optionId);
+        registry.category("website-plugins").remove(pluginId);
     });
 }
+function makeOptionPlugin({ id, template, selector, sequence, OptionComponent }) {
```
Comment (FrancoisGe) [inline addons/html_builder/static/tests/helpers.js:151]: exclude

File: addons/html_builder/static/src/builder/options/content_width_option.js [hunk 6]
```diff
@@ -1,11 +1,18 @@
...
-});
+class ContentWidthOptionPlugin extends Plugin {
+    static id = "ContentWidthOption";
+    resources = {
+        builder_options: [
+            {
+                template: "html_builder.ContentWidthOption",
+                selector: "section, .s_carousel .carousel-item, .s_carousel_intro_item",
+                // TODO add exclude and target and remove applyTo in the template of ContentWidthOption
+                // exclude: "[data-snippet] :not(.oe_structure) > [data-snippet]",
```
Comment (FrancoisGe) [inline addons/html_builder/static/src/builder/options/content_width_option.js]: exclude

File: addons/html_builder/static/src/builder/options/alignment_option.js [hunk 7]
```diff
@@ -1,6 +1,15 @@
...
-    template: "html_builder.AlignmentOption",
-    selector: ".s_share, .s_text_highlight, .s_social_media, section",
-});
+class AlignmentOptionPlugin extends Plugin {
+    static id = "AlignmentOption";
+    resources = {
+        builder_options: [
+            {
+                template: "html_builder.AlignmentOption",
+                selector: ".s_share, .s_text_highlight, .s_social_media, section",
```
Comment (FrancoisGe) [inline addons/html_builder/static/src/builder/options/alignment_option.js]: remove section

File: addons/html_builder/static/src/builder/options/border_option.js [hunk 8]
```diff
@@ -1,6 +1,15 @@
...
-    template: "html_builder.BorderOption",
-    selector: "section .row > div", // TODO to use the correct selector
-});
+class BorderOptionPlugin extends Plugin {
+    static id = "BorderOption";
+    resources = {
+        builder_options: [
+            {
+                template: "html_builder.BorderOption",
+                selector: "section .row > div", // TODO to use the correct selector
```
Comment (FrancoisGe) [inline addons/html_builder/static/src/builder/options/border_option.js]: exclude

File: addons/html_builder/static/src/builder/options/layout_option.js [hunk 9]
```diff
@@ -1,15 +1,31 @@
...
     toggleGridMode,
 } from "@html_builder/builder/utils/grid_layout_utils";
 
+class LayoutOptionPlugin extends Plugin {
+    static id = "LayoutOption";
+    resources = {
+        builder_options: {
+            OptionComponent: LayoutOption,
+            selector: "section, section.s_carousel_wrapper .carousel-item, .s_carousel_intro_item",
+            // TODO add exclude  data-exclude=".s_dynamic, .s_dynamic_snippet_content, .s_dynamic_snippet_title, .s_masonry_block, .s_framed_intro, .s_features_grid, .s_media_list, .s_table_of_content, .s_process_steps, .s_image_gallery, .s_timeline, .s_pricelist_boxed, .s_quadrant, .s_pricelist_cafe, .s_faq_horizontal, .s_image_frame, .s_card_offset, .s_contact_info"
```
Comment (FrancoisGe) [inline addons/html_builder/static/src/builder/options/layout_option.js]: exclude

**Pull request merged**